### ISSUE TITLE
RFC: Split Music Player Into Separate Machine

### ISSUE THREAD

**Issue by zachreizner** [acm-uiuc/beats#20]
This is related to implementing #13 and #15. I propose that we write a thin client that connects to the main server and identifies itself as, for example, room 1404. The server would then respond to the client the item to play along with a url for it. The client would then play the item as well as drive the display for playing YouTube or album art. The backend would keep multiple queues, one for each client. The backend/client connection would persist for playback control purposes, but the client would determine when a song has completed and the queue should be advanced.

Advantage to this architecture is that this would make the playback devices very cheap allowing many instances i.e. rooms. The users would be connecting to the ACM servers which are powerful enough to handle multiple concurrent users. This means a smoother web experience for users. Taking the user traffic away from the playback devices prevents awful sound glitches that happen on heavily loaded machines. The downside is that this complicates the architecture of the project and would likely increase latency of the whole system.

**TL;DR**
Thin music/video playback device that connects to server to play songs from the queue for a room

**Side Note**
This change also has the unrelated advantage of allowing adhoc rooms using somebody's laptop as a playback device, perhaps connected to the speakers of room 2405. I'm pretty sure something like this already exists, but it would be nice to unify the web interfaces and music libraries.

Any comments? @CJS7070 for network purposes. @kevinwang for design considerations.
